cookbooks/aws-parallelcluster-slurm/files/default/config_slurm/scripts/health_checks/gpu_health_check.sh

```diff
@@ -0,0 +1,209 @@
+
+  ## Check if the instance has an NVIDIA acceleration
+  is_nvidia
+  fast_success_exit "$?" "The GPU Health Check is running in an instance without GPU: skipping its execution."
+
+  ## Check if there are GPU associated to the job
+  if [ -z "$GPU_DEVICE_ORDINAL" ]; then
+#    log_info "The GPU Health Check is not required since there are no GPU associated to the job."
+#    return 0
+    nvidia_smi_output=$(nvidia-smi -L)
+    fast_success_exit "$?" "The GPU Health Check is running in an instance without nvidia-smi available: skipping its execution."
+    GPU_DEVICE_ORDINAL=$(echo "$nvidia_smi_output" | while IFS= read -r gpu ; do echo $gpu | cut -d ':' -f1 | sed s/"GPU "//; done | tr '\n' ',' | sed 's/.$//')
+  fi
+
+  ## Services Initialization
+  local dcgm_service_started
+  initialize_dcgm
+  fast_success_exit "$?" "skipping execution of the GPU Health Check"
+
+  local nvidia_fabricmanager_service_started
+  initialize_nvidia_fabricmanager
+  fast_success_exit "$?" "Skipping execution of the GPU Health Check"
+
+  ## Retrieve gpus information and enable them to run the health check
+  local gpus_list
+  gpus_list=${GPU_DEVICE_ORDINAL//,/$'\n'}
+  for gpu_id in $gpus_list
+  do
+    # Retrieve gpu information. This check will exit in following cases:
+    # - the nvidia-smi command is not available
+    # - the nvidia-smi command is available but it fails its execution
+    nvidia_smi_out=$(nvidia-smi -q -i $gpu_id)
+    nvidia_smi_exit_code=$(echo $?)
+    if [ $nvidia_smi_exit_code -eq 127 ]; then
+      fast_success_exit $nvidia_smi_exit_code "Health check has been executed but the nvidia-smi tool is not available in this system or is located in a different file path. If possible use an official Parallel Cluster AMI."
+    fi
+    fast_success_exit $nvidia_smi_exit_code "The nvidia-smi tool fails its execution."
+
+    ## Enable persistence mode in target GPU, required by the DCGM Diagnostic tool
+    log_info "Enabling persistence mode in target gpu {$gpu_id} if not already enabled"
+    local persistence_mode_enabled_$gpu_id
+    persistence_mode=$(echo "$nvidia_smi_out" | grep "Persistence Mode")
+    if [[ ! ${persistence_mode} == *"Enabled"* ]]; then
+      sudo nvidia-smi -i $gpu_id -pm 1
+      declare persistence_mode_enabled_$gpu_id=true
+    fi
+
+    ## Log useful GPU information
+    log_info "GPU : $gpu_id"
+    log_info "$(echo "$nvidia_smi_out" | grep "GPU UUID")"
+    log_info "$(echo "$nvidia_smi_out" | grep "Serial Number")"
+  done
+
+  ## Run GPUs Health Check test
+  log_info "Running GPU Health Check tool"
+  dcgmi diag -i $GPU_DEVICE_ORDINAL -r 2
+  dcgmi_exit_code=$(echo $?)
+
+  if [ $dcgmi_exit_code -ne 0 ]; then
+    if [ $dcgmi_exit_code -eq 127 ]; then
+      log_info "Health check has been executed but the DCGM Diagnostic tool is not available in this system or is located in a different file path. If possible use an official Parallel Cluster AMI."
```

Comment:
can we move the check on the existence of the dcgmi tool at the beggining of the scripts, e.g. in a prerequisite section, for example together with the check of the nvidia-smi precence. Instead of calling the command we can use other builtin method to verify if the command exist and it's executable. Then here just fail if execit code is different from 0